### PR TITLE
fix <noname> folder

### LINKED ISSUE #14: <no-name> directory in Azure Portal
New blob e.g.:
https://URL/azmi-test//tmp/azmi_integration_test_2020-01-09_10:23:12.txt

Notice double slash between azmi-test and tmp.
It generates <no-name> directory in Azure Portal.

### PR DESCRIPTION
**Pull request by iricigor**
- Added fix by trimming left slash for blob name
- Added tests which upload file with absolute path

Resolves #14 

Potential further issues:
- uploading file with path `./myname`
- uploading file with path `../dir/name`

### CODE REVIEW

**Comment by iricigor** [comment]
New build and integration test on the branch were successful.

File: test/integration/integration_test.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ test "Should fail: Save file contents to read-only Azure storage container" asse
 # Role(s):    Storage Blob Data Contributor
 # Profile(s): bt-seu-test-id (obj. ID: d1c05b65-ccf9-47bd-870d-4e44d209ee7a), kotipoiss-identity (obj. ID: ccb781af-a4eb-4ecc-b183-cef74b3cc717)
 CONTAINER_URL="https://azmitest.blob.core.windows.net/azmi-itest-rw"
-TIMESTAMP=`date "+%Y-%m-%d_%H:%M:%S"` # e.g. 2020-01-07_14:41:02
+TIMESTAMP=`date "+%Y%m%d_%H%M%S"` # e.g. 2020-01-07_14:41:02
```
Comment (MangyCoyote) [inline test/integration/integration_test.sh]: Edit example.

File: test/integration/integration_test.sh [hunk 2]
```diff
@@ -87,6 +87,12 @@ RANDOM_BLOB_TO_STORE_SHA256=$(sha256sum $RANDOM_BLOB_TO_STORE | awk '{ print $1
 DOWNLOADED_BLOB_SHA256=$(sha256sum $DOWNLOADED_BLOB | awk '{ print $1 }')
 test "Blobs have to have equal SHA256 checksums" assert.Success "[ $RANDOM_BLOB_TO_STORE_SHA256 = $DOWNLOADED_BLOB_SHA256 ]"
 
+# there should be no <noname> folder in Azure
+testing class "noname"
+test "Prepare tmp file" assert.Success "rm -f /tmp/${RANDOM_BLOB_TO_STORE} && echo sometext > /tmp/${RANDOM_BLOB_TO_STORE}"
+test "Upload tmp file" assert.Success "azmi setblob -f /tmp/${RANDOM_BLOB_TO_STORE} --container ${CONTAINER_URL}"
+test "There is no noname folder after upload" assert.Fail "azmi getblob -f /dev/null -b ${CONTAINER_URL}//tmp/${RANDOM_BLOB_TO_STORE}"
```
Comment (MangyCoyote) [inline test/integration/integration_test.sh]: Have you tested it?

Comment (iricigor) [reply test/integration/integration_test.sh]: yes

**Pull request merged**